### PR TITLE
Daily Chest Implementation

### PR DESCRIPTION
**Pull request by RealistikDash**
This PR aims at implementing the Daily Chest feature into RealistikGDPS.

## Objectives
- [x] Realistic loot tables for both types of chests
- [x] Keep track of reward history
- [x] Reward milestones (key every 500 mana)
- [x] Geometry Dash endpoint
- [x] Full Testing

### CODE REVIEW

**Comment by RealistikDash** [comment]
The endpoint currently is broken (stuck at this screen).
image

Had to be one of the worst endpoints for how messed up the responses were and how many responsibilities it has.

File: rgdps/common/gd_obj.py [hunk 1]
```diff
@@ -274,3 +277,67 @@ def comma_separated_ints(data: str) -> list[int]:
...
+
+def joined_string(*data: SupportsStr, separator: str = ",") -> str:
+    return separator.join(str(x) for x in data)
+
+
+def create_chest_rewards(chest: DailyChest) -> str:
+    return joined_string(
+        chest.mana,
+        chest.diamonds,
+        0,  # TODO: Shards,
```
Comment (RealistikDash) [inline rgdps/common/gd_obj.py:291]: TODO: Figure out how the shard data works.

**Comment by RealistikDash** [comment]
Note to self: Demon keys may be entirely client side. The demon key value may be something completely else used for multiple shards.
https://github.com/Wyliemaster/GD-Decompiled/blob/bc41c450ba09033ac4592622d739260262085d0a/GD/code/src/GJRewardItem.cpp#L39-L89

**Comment by RealistikDash** [comment]
image
It works!

**Review by tsunyoku — APPROVED** [review]

**Pull request merged**